### PR TITLE
Feature/#212 Add QueryHandler Dataflow Abstraction

### LINKED ISSUE #212: Add QueryHandler Dataflow Abstraction
* All files below directory [logistic_regression/exasol_machine_learning_library/execution/sql_stage_graph_execution](https://github.com/exasol/machine-learning-library/tree/logistic_regression/exasol_machine_learning_library/execution/sql_stage_graph_execution)
* plus Dependencies in directory `execution/stage_graph/`
  * [sql_stage_train_query_handler.py](https://github.com/exasol/machine-learning-library/blob/logistic_regression/exasol_machine_learning_library/execution/stage_graph/sql_stage_train_query_handler.py)
  * [stage_graph.py](https://github.com/exasol/machine-learning-library/blob/logistic_regression/exasol_machine_learning_library/execution/stage_graph/stage_graph.py)
  * [stage.py](https://github.com/exasol/machine-learning-library/blob/logistic_regression/exasol_machine_learning_library/execution/stage_graph/stage.py) without class `UDFStage`
* plus related Tests in [tests/unit_tests/sql_stage_graph/](https://github.com/exasol/machine-learning-library/blob/logistic_regression/tests/unit_tests/sql_stage_graph) and subdirectories
* After that some small adaption will be required

### LINKED ISSUE #216: Rename MLL modules
Ticket #212 requested to move parts of MLL repository to AAF including unit tests.

As a cleanup step, the current ticket requests to rename the modules:

image

### PR DESCRIPTION
**Pull request by ckunki**
Closes #212 
Closes #216

### CODE REVIEW

File: exasol_machine_learning_library/execution/stage_graph/stage.py [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+ParameterType = TypeVar("ParameterType", bound=Parameter)
+ResultType = TypeVar("ResultType", bound=Result)
+
+
+class Stage(ABC, Generic[ParameterType, ResultType]):
+    pass
+
+
+class SQLStage(Stage):
```
Comment (tkilias) [inline exasol_machine_learning_library/execution/stage_graph/stage.py:19]: extract

File: exasol_machine_learning_library/execution/stage_graph/stage.py [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    @abc.abstractmethod
+    def create_train_query_handler(self,
+                                   stage_input: SQLStageTrainQueryHandlerInput,
+                                   query_handler_context: ScopeQueryHandlerContext) \
+            -> SQLStageTrainQueryHandler:
+        pass
+
+
+class UDFStage(Stage):
```
[the remark this answers is not in the document]
Comment (ckunki) [reply exasol_machine_learning_library/execution/stage_graph/stage.py:28]: `execution/sql_stage_graph/sql_stage_graph.py` curently imports `UDFStage`.

[the remark this answers is not in the document]
Comment (ckunki) [reply exasol_machine_learning_library/execution/stage_graph/stage.py:28]: I removed all classes from`execution/sql_stage_graph/sql_stage_graph.py` with references to `UDFStage`:
* `UDFRunnerPlaceholderSQLStage`
* `UDFRunnerSQLStage`
* `ColumnSelectorPlaceholderUDFStage`
* `SourceUDFStage`
* `SinkUDFStage`

File: exasol_machine_learning_library/execution/stage_graph/stage.py (hunk 1 above)
Comment (ckunki) [reply exasol_machine_learning_library/execution/stage_graph/stage.py:19]: Moved to separate file `sql_stage.py`

File: doc/changes/changes_0.1.1.md [hunk 3]
```diff
@@ -4,6 +4,10 @@ Code name:
 
 ## Summary
 
+## Features
+
+* #212: Added QueryHandler Dataflow Abstraktion
```
Comment (tkilias) [inline doc/changes/changes_0.1.1.md:9]:
```suggestion
* #212: Added QueryHandler Dataflow Abstraction
```

Comment (tkilias) [reply doc/changes/changes_0.1.1.md:9]: I am almost thinking, maybe, we should add a spellchecker to the CI

File: exasol/analytics/query_handler/execution_graph.py
Comment (tkilias) [inline exasol/analytics/query_handler/execution_graph.py:1]: Move to exasol/analytics/query_handler/graph/execution_graph.py

File: exasol/analytics/query_handler/graph/stage/sql/execution/input_output.py
Comment (tkilias) [inline exasol/analytics/query_handler/graph/stage/sql/execution/input_output.py:1]: Move to exasol/analytics/query_handler/graph/stage/sql/input_output.py

File: exasol/analytics/query_handler/graph/stage/sql/execution/data_partition.py
Comment (tkilias) [inline exasol/analytics/query_handler/graph/stage/sql/execution/data_partition.py:1]: Move to exasol/analytics/query_handler/graph/stage/sql/data_partition.py

File: exasol/analytics/query_handler/graph/stage/sql/execution/dataset.py
Comment (tkilias) [inline exasol/analytics/query_handler/graph/stage/sql/execution/dataset.py:1]: Move to exasol/analytics/query_handler/graph/stage/sql/dataset.py

File: exasol/analytics/query_handler/graph/stage/sql/execution/dependency.py
Comment (tkilias) [inline exasol/analytics/query_handler/graph/stage/sql/execution/dependency.py:1]: Move to exasol/analytics/query_handler/graph/stage/sql/dependency.py

File: tests/unit_tests/sql_stage_graph/mock_cast.py
Comment (tkilias) [inline tests/unit_tests/sql_stage_graph/mock_cast.py:1]: exasol/advanced-analytics-framework/tests/mock_cast.py

Comment (ckunki) [reply tests/unit_tests/sql_stage_graph/mock_cast.py:1]: See #217

File: exasol/analytics/query_handler/graph/stage/sql/execution/dependency.py
Comment (ckunki) [reply exasol/analytics/query_handler/graph/stage/sql/execution/dependency.py:1]: See #217

File: exasol/analytics/query_handler/graph/stage/sql/execution/dataset.py
Comment (ckunki) [reply exasol/analytics/query_handler/graph/stage/sql/execution/dataset.py:1]: See #217

File: exasol/analytics/query_handler/graph/stage/sql/execution/data_partition.py
Comment (ckunki) [reply exasol/analytics/query_handler/graph/stage/sql/execution/data_partition.py:1]: See #217

File: exasol/analytics/query_handler/graph/stage/sql/execution/input_output.py
Comment (ckunki) [reply exasol/analytics/query_handler/graph/stage/sql/execution/input_output.py:1]: See #217

File: exasol/analytics/query_handler/execution_graph.py
Comment (ckunki) [reply exasol/analytics/query_handler/execution_graph.py:1]: See #217

File: exasol_machine_learning_library/execution/stage_graph/stage.py (hunk 1 above)
Comment (ckunki) [reply exasol_machine_learning_library/execution/stage_graph/stage.py]: See #217

**Review by tkilias — APPROVED** [review]

**Pull request merged**